### ISSUE TITLE
Long sync with peer

### ISSUE THREAD

**Issue by JiapengLi** [GenesysGo/dagger-testnet-issues#15]
```
2024-01-08T03:08:43.144290376+00:00 [INFO] dagger::processor - entering processor loop
2024-01-08T03:08:43.144557071+00:00 [INFO] dagger::client - entering sync loop
2024-01-08T03:08:43.144720990+00:00 [INFO] dagger::client - entering conn loop

=> stuck for about 9 hours

2024-01-08T03:17:10.200642807+00:00 [WARN] dagger::client - Long sync with peer Cb1CwnBppUWfNZmeZfJoVdVzsbrJXkTHN7Fk64Q82dwy: 5626 millis
```

```
2024-01-08T03:08:43.144290376+00:00 [INFO] dagger::processor - entering processor loop
2024-01-08T03:08:43.144557071+00:00 [INFO] dagger::client - entering sync loop
2024-01-08T03:08:43.144720990+00:00 [INFO] dagger::client - entering conn loop
2024-01-08T03:17:10.200642807+00:00 [WARN] dagger::client - Long sync with peer Cb1CwnBppUWfNZmeZfJoVdVzsbrJXkTHN7Fk64Q82dwy: 5626 millis
2024-01-08T03:17:13.974522579+00:00 [WARN] dagger::client - Long sync with peer ARfg7k3wTR43W9pqNhNVsSAvazDfvXyNNNVyKjn4GwaN: 5436 millis
2024-01-08T04:10:44.823034406+00:00 [WARN] dagger::client - Long sync with peer EzxFgswVDu9S9sVZ1hM137AuqRvXudUdWzsa6t8tBEBj: 5286 millis
2024-01-08T04:10:47.138977490+00:00 [WARN] dagger::client - Long sync with peer 9WDbcQG5yvg2VAFXiSAnJoAbnD4D3SMx1YQUrLiUHvKy: 5700 millis
2024-01-08T04:10:51.848220910+00:00 [WARN] dagger::client - Long sync with peer Cb1CwnBppUWfNZmeZfJoVdVzsbrJXkTHN7Fk64Q82dwy: 6624 millis
2024-01-08T04:10:57.540575342+00:00 [ERROR] dagger::client - timeout syncing with peer ARfg7k3wTR43W9pqNhNVsSAvazDfvXyNNNVyKjn4GwaN
2024-01-08T04:11:00.755713680+00:00 [WARN] dagger::client - Long sync with peer Cb1CwnBppUWfNZmeZfJoVdVzsbrJXkTHN7Fk64Q82dwy: 8505 millis
2024-01-08T04:11:05.512213394+00:00 [WARN] dagger::client - Long sync with peer ARfg7k3wTR43W9pqNhNVsSAvazDfvXyNNNVyKjn4GwaN: 7971 millis
2024-01-08T04:45:14.208933603+00:00 [ERROR] dagger::client - timeout syncing with peer EzxFgswVDu9S9sVZ1hM137AuqRvXudUdWzsa6t8tBEBj
2024-01-08T04:45:15.999943006+00:00 [ERROR] dagger::client - timeout syncing with peer 9WDbcQG5yvg2VAFXiSAnJoAbnD4D3SMx1YQUrLiUHvKy
2024-01-08T04:45:29.886272711+00:00 [WARN] dagger::client - Long sync with peer 9WDbcQG5yvg2VAFXiSAnJoAbnD4D3SMx1YQUrLiUHvKy: 6020 millis
2024-01-08T04:51:15.225283392+00:00 [WARN] dagger::client - Long sync with peer 9WDbcQG5yvg2VAFXiSAnJoAbnD4D3SMx1YQUrLiUHvKy: 8250 millis
2024-01-08T04:51:16.853763519+00:00 [WARN] dagger::client - Long sync with peer ARfg7k3wTR43W9pqNhNVsSAvazDfvXyNNNVyKjn4GwaN: 9066 millis
2024-01-08T04:55:18.854469040+00:00 [WARN] dagger::client - Long sync with peer 9WDbcQG5yvg2VAFXiSAnJoAbnD4D3SMx1YQUrLiUHvKy: 8440 millis
2024-01-08T04:55:25.872711632+00:00 [WARN] dagger::client - Long sync with peer EzxFgswVDu9S9sVZ1hM137AuqRvXudUdWzsa6t8tBEBj: 7754 millis
2024-01-08T05:55:58.430471018+00:00 [WARN] dagger::client - Long sync with peer EzxFgswVDu9S9sVZ1hM137AuqRvXudUdWzsa6t8tBEBj: 6052 millis
2024-01-08T05:56:12.106153061+00:00 [WARN] dagger::client - Long sync with peer 6uFd3N1H3bd7XA2bsRRFXYqTqMKY6MPzxdpocS36eDa6: 6603 millis
2024-01-08T06:01:01.613929169+00:00 [WARN] dagger::client - Long sync with peer EzxFgswVDu9S9sVZ1hM137AuqRvXudUdWzsa6t8tBEBj: 5558 millis
2024-01-08T06:01:07.003354967+00:00 [WARN] dagger::client - Long sync with peer Cb1CwnBppUWfNZmeZfJoVdVzsbrJXkTHN7Fk64Q82dwy: 8424 millis
2024-01-08T06:01:17.367236765+00:00 [WARN] dagger::client - Long sync with peer ARfg7k3wTR43W9pqNhNVsSAvazDfvXyNNNVyKjn4GwaN: 7778 millis
2024-01-08T06:01:20.949305841+00:00 [WARN] dagger::client - Long sync with peer Cb1CwnBppUWfNZmeZfJoVdVzsbrJXkTHN7Fk64Q82dwy: 6108 millis
2024-01-08T06:06:10.598115141+00:00 [ERROR] dagger::client - timeout syncing with peer EzxFgswVDu9S9sVZ1hM137AuqRvXudUdWzsa6t8tBEBj
2024-01-08T06:06:12.685569353+00:00 [ERROR] dagger::client - timeout syncing with peer 9WDbcQG5yvg2VAFXiSAnJoAbnD4D3SMx1YQUrLiUHvKy
2024-01-08T06:06:15.770640194+00:00 [WARN] dagger::client - Long sync with peer 6uFd3N1H3bd7XA2bsRRFXYqTqMKY6MPzxdpocS36eDa6: 5172 millis
2024-01-08T06:06:24.769428079+00:00 [WARN] dagger::client - Long sync with peer 9WDbcQG5yvg2VAFXiSAnJoAbnD4D3SMx1YQUrLiUHvKy: 7739 millis
2024-01-08T06:10:13.805672474+00:00 [WARN] dagger::client - Long sync with peer 9WDbcQG5yvg2VAFXiSAnJoAbnD4D3SMx1YQUrLiUHvKy: 8301 millis
2024-01-08T06:10:16.554771218+00:00 [WARN] dagger::client - Long sync with peer ARfg7k3wTR43W9pqNhNVsSAvazDfvXyNNNVyKjn4GwaN: 5175 millis
2024-01-08T06:15:13.865193036+00:00 [WARN] dagger::client - Long sync with peer 6uFd3N1H3bd7XA2bsRRFXYqTqMKY6MPzxdpocS36eDa6: 5435 millis
2024-01-08T06:15:19.429761432+00:00 [WARN] dagger::client - Long sync with peer 9WDbcQG5yvg2VAFXiSAnJoAbnD4D3SMx1YQUrLiUHvKy: 9058 millis
2024-01-08T06:15:30.829458720+00:00 [WARN] dagger::client - Long sync with peer EzxFgswVDu9S9sVZ1hM137AuqRvXudUdWzsa6t8tBEBj: 7483 millis
2024-01-08T07:16:58.221037034+00:00 [ERROR] dagger::client - timeout syncing with peer Cb1CwnBppUWfNZmeZfJoVdVzsbrJXkTHN7Fk64Q82dwy
2024-01-08T07:23:24.703575931+00:00 [ERROR] dagger::client - timeout syncing with peer 6uFd3N1H3bd7XA2bsRRFXYqTqMKY6MPzxdpocS36eDa6
2024-01-08T07:23:24.769213607+00:00 [ERROR] dagger::client - timeout syncing with peer 9WDbcQG5yvg2VAFXiSAnJoAbnD4D3SMx1YQUrLiUHvKy
2024-01-08T07:23:34.704621882+00:00 [ERROR] dagger_p2p - failed connect MaxHandshakeDurationExceeded { max_handshake_duration: 10s, source: Location { file: "/.cargo/registry/src/index.crates.io-6f17d22bba15001f/s2n-quic-transport-0.31.0/src/connection/connection_impl.rs", line: 1053, col: 24 } }
2024-01-08T07:23:34.769729540+00:00 [ERROR] dagger_p2p - failed connect MaxHandshakeDurationExceeded { max_handshake_duration: 10s, source: Location { file: "/.cargo/registry/src/index.crates.io-6f17d22bba15001f/s2n-quic-transport-0.31.0/src/connection/connection_impl.rs", line: 1053, col: 24 } }
2024-01-08T07:23:44.706270819+00:00 [ERROR] dagger_p2p - failed connect MaxHandshakeDurationExceeded { max_handshake_duration: 10s, source: Location { file: "/.cargo/registry/src/index.crates.io-6f17d22bba15001f/s2n-quic-transport-0.31.0/src/connection/connection_impl.rs", line: 1053, col: 24 } }
2024-01-08T07:23:44.770778615+00:00 [ERROR] dagger::client - timeout syncing with peer Cb1CwnBppUWfNZmeZfJoVdVzsbrJXkTHN7Fk64Q82dwy
2024-01-08T07:23:55.479921023+00:00 [WARN] dagger::client - Long sync with peer EzxFgswVDu9S9sVZ1hM137AuqRvXudUdWzsa6t8tBEBj: 8052 millis
2024-01-08T09:06:47.538814010+00:00 [ERROR] dagger::client - timeout syncing with peer ARfg7k3wTR43W9pqNhNVsSAvazDfvXyNNNVyKjn4GwaN
2024-01-08T09:06:47.539196578+00:00 [ERROR] dagger_p2p - failed connect MaxHandshakeDurationExceeded { max_handshake_duration: 10s, source: Location { file: "/.cargo/registry/src/index.crates.io-6f17d22bba15001f/s2n-quic-transport-0.31.0/src/connection/connection_impl.rs", line: 1053, col: 24 } }
2024-01-08T09:06:57.539491153+00:00 [ERROR] dagger::client - timeout syncing with peer EzxFgswVDu9S9sVZ1hM137AuqRvXudUdWzsa6t8tBEBj
2024-01-08T09:06:57.539761674+00:00 [ERROR] dagger_p2p - failed connect MaxHandshakeDurationExceeded { max_handshake_duration: 10s, source: Location { file: "/.cargo/registry/src/index.crates.io-6f17d22bba15001f/s2n-quic-transport-0.31.0/src/connection/connection_impl.rs", line: 1053, col: 24 } }
2024-01-08T09:07:07.541067362+00:00 [ERROR] dagger_p2p - failed connect MaxHandshakeDurationExceeded { max_handshake_duration: 10s, source: Location { file: "/.cargo/registry/src/index.crates.io-6f17d22bba15001f/s2n-quic-transport-0.31.0/src/connection/connection_impl.rs", line: 1053, col: 24 } }
2024-01-08T09:07:07.541222484+00:00 [ERROR] dagger_p2p - failed connect MaxHandshakeDurationExceeded { max_handshake_duration: 10s, source: Location { file: "/.cargo/registry/src/index.crates.io-6f17d22bba15001f/s2n-quic-transport-0.31.0/src/connection/connection_impl.rs", line: 1053, col: 24 } }
2024-01-08T10:16:40.022897294+00:00 [ERROR] dagger::client - timeout syncing with peer ARfg7k3wTR43W9pqNhNVsSAvazDfvXyNNNVyKjn4GwaN
2024-01-08T10:16:40.037271266+00:00 [ERROR] dagger::client - timeout syncing with peer Cb1CwnBppUWfNZmeZfJoVdVzsbrJXkTHN7Fk64Q82dwy
2024-01-08T10:16:50.023833418+00:00 [ERROR] dagger::client - timeout syncing with peer EzxFgswVDu9S9sVZ1hM137AuqRvXudUdWzsa6t8tBEBj
2024-01-08T10:16:50.880985247+00:00 [ERROR] dagger::client - timeout syncing with peer ARfg7k3wTR43W9pqNhNVsSAvazDfvXyNNNVyKjn4GwaN
2024-01-08T10:17:00.026058376+00:00 [ERROR] dagger::client - timeout syncing with peer 9WDbcQG5yvg2VAFXiSAnJoAbnD4D3SMx1YQUrLiUHvKy
2024-01-08T10:17:00.881750065+00:00 [ERROR] dagger_p2p - failed connect MaxHandshakeDurationExceeded { max_handshake_duration: 10s, source: Location { file: "/.cargo/registry/src/index.crates.io-6f17d22bba15001f/s2n-quic-transport-0.31.0/src/connection/connection_impl.rs", line: 1053, col: 24 } }
2024-01-08T10:17:06.328693262+00:00 [WARN] dagger::client - Long sync with peer EzxFgswVDu9S9sVZ1hM137AuqRvXudUdWzsa6t8tBEBj: 5446 millis
2024-01-08T10:17:07.442904094+00:00 [WARN] dagger::client - Long sync with peer 6uFd3N1H3bd7XA2bsRRFXYqTqMKY6MPzxdpocS36eDa6: 7416 millis

```

**Comment by OlajideOgun** [comment]
I am experiencing the same issue dont want to create another tickets but here are my logs

```Jan 11 01:14:51 taiwebapp start_wield.sh[420490]: joining in on epoch 8754
Jan 11 01:14:51 taiwebapp start_wield.sh[420490]: 2024-01-11T01:14:51.688501470+00:00 [INFO] dagger::p2p::join - Attempting to join network with 'trusted' peer: 184.1>
Jan 11 01:14:51 taiwebapp start_wield.sh[420490]: 2024-01-11T01:14:51.823239644+00:00 [INFO] dagger::p2p::join - Connected to trusted peer
Jan 11 01:14:51 taiwebapp start_wield.sh[420490]: 2024-01-11T01:14:51.823314275+00:00 [INFO] dagger::p2p::join - Requested peer list from trusted peer
Jan 11 01:14:52 taiwebapp start_wield.sh[420490]: 2024-01-11T01:14:52.090900196+00:00 [INFO] dagger::p2p::join - Obtained peer list from trusted peer
Jan 11 01:14:52 taiwebapp start_wield.sh[420490]: 2024-01-11T01:14:52.097909694+00:00 [INFO] dagger::p2p::join - Successfully joined network
Jan 11 01:14:52 taiwebapp start_wield.sh[420490]: 2024-01-11T01:14:52.098148901+00:00 [INFO] dagger::p2p::graph::dagger - entering dagger loop
Jan 11 01:14:52 taiwebapp start_wield.sh[420490]: 2024-01-11T01:14:52.098608436+00:00 [INFO] dagger::controller - entering controller loop
Jan 11 01:14:52 taiwebapp start_wield.sh[420490]: 2024-01-11T01:14:52.101842302+00:00 [INFO] dagger::client - joined with peer list {Peer { pubkey: 6Fv9bYXVK6HgspzJye>
Jan 11 01:14:52 taiwebapp start_wield.sh[420490]: 2024-01-11T01:14:52.105036844+00:00 [INFO] dagger::client - entering conn loop
Jan 11 01:14:52 taiwebapp start_wield.sh[420490]: 2024-01-11T01:14:52.105215804+00:00 [INFO] dagger::client - entering sync loop
Jan 11 01:14:52 taiwebapp start_wield.sh[420490]: 2024-01-11T01:14:52.106213622+00:00 [INFO] dagger::processor - entering processor loop
Jan 11 04:08:04 taiwebapp start_wield.sh[420490]: 2024-01-11T04:08:04.049270756+00:00 [ERROR] dagger::client - timeout syncing with peer EzxFgswVDu9S9sVZ1hM137AuqRvXu>
Jan 11 04:08:15 taiwebapp start_wield.sh[420490]: 2024-01-11T04:08:15.199734365+00:00 [ERROR] dagger::client - timeout syncing with peer EzxFgswVDu9S9sVZ1hM137AuqRvXu>
Jan 11 04:08:28 taiwebapp start_wield.sh[420490]: 2024-01-11T04:08:28.889292772+00:00 [WARN] dagger::client - Long sync with peer EzxFgswVDu9S9sVZ1hM137AuqRvXudUdWzsa>
Jan 11 04:08:41 taiwebapp start_wield.sh[420490]: 2024-01-11T04:08:41.974356138+00:00 [ERROR] dagger::client - timeout syncing with peer EzxFgswVDu9S9sVZ1hM137AuqRvXu>
Jan 12 10:30:56 taiwebapp start_wield.sh[420490]: 2024-01-12T10:30:56.897886415+00:00 [WARN] dagger::client - Long sync with peer 9WDbcQG5yvg2VAFXiSAnJoAbnD4D3SMx1YQU>
Jan 13 21:01:27 taiwebapp start_wield.sh[420490]: 2024-01-13T21:01:27.209276037+00:00 [WARN] dagger::client - Long sync with peer Cb1CwnBppUWfNZmeZfJoVdVzsbrJXkTHN7Fk>
Jan 14 12:54:02 taiwebapp start_wield.sh[420490]: 2024-01-14T12:54:02.428124356+00:00 [WARN] dagger::client - Long sync with peer 9WDbcQG5yvg2VAFXiSAnJoAbnD4D3SMx1YQU>```

**Comment by JiapengLi** [comment]
@OlajideOgun 

The issue appears to have been related to the network quality of the wield node. It's been observed that many operators choose Contabo for running their nodes, possibly due to the lowwer price, especially in the German region. This might be a good option for you to consider.

Regarding the Long Sync issue with wield nodes, it seems they rarely recover on their own. A solution to try is restarting the wield node and then observing it for about 15 minutes. From my experience, if the process is successful, you should see messages like "bundle xxxx finalized!" indicating that the node is functioning properly.

**Comment by asln82Ns** [comment]
Got an issue with max handshakes then the node completely stopped. After reboot I got entering conn loop then nothing. I deleted the first 3 trusted peers from config.toml and it started working again. attached is the config log.


[config.log](https://github.com/GenesysGo/dagger-testnet-issues/files/13940376/config.log)

**Comment by OlajideOgun** [comment]
Which provider were you using before I am using Webdock premium node. Are you saying only contabo works?

>

**Comment by OlajideOgun** [comment]
my issue is fixed in 0.7.4.
